### PR TITLE
Update VPA to v0.14.0

### PR DESCRIPTION
**Pull request by voelzmo**
**How to categorize this PR?**

/area auto-scaling
/kind enhancement

**What this PR does / why we need it**:
Update VPA to 0.14.0 to get newest fixes. 

See entire release notes: https://github.com/kubernetes/autoscaler/releases/tag/vertical-pod-autoscaler-0.14.0

Removed the test for CRD content, as it was annoying and pure copy-paste from production templates.

**Release note**:

```other operator
Update `vertical-pod-autoscaler` to `v0.14.0`.
```

### CODE REVIEW

File: pkg/component/vpa/vpa_test.go [hunk 1]
```diff
@@ -1570,8 +1570,6 @@ var _ = Describe("VPA", func() {
 				Expect(string(managedResourceSecret.Data["clusterrole____gardener.cloud_vpa_target_target-reader.yaml"])).To(Equal(componenttest.Serialize(clusterRoleGeneralTargetReader)))
 				Expect(string(managedResourceSecret.Data["clusterrolebinding____gardener.cloud_vpa_target_target-reader.yaml"])).To(Equal(componenttest.Serialize(clusterRoleBindingGeneralTargetReader)))
 				Expect(string(managedResourceSecret.Data["mutatingwebhookconfiguration____vpa-webhook-config-target.yaml"])).To(Equal(componenttest.Serialize(mutatingWebhookConfiguration)))
-				Expect(string(managedResourceSecret.Data["crd-verticalpodautoscalercheckpoints.yaml"])).To(Equal(crdVPACheckpoints))
-				Expect(string(managedResourceSecret.Data["crd-verticalpodautoscalers.yaml"])).To(Equal(crdVPA))
```
Comment (rfranzke) [inline pkg/component/vpa/vpa_test.go:1574]: We should at least assert that the CRDs get deployed

```suggestion
				Expect(managedResourceSecret.Data).To(HaveKey("crd-verticalpodautoscalercheckpoints.yaml"))
				Expect(managedResourceSecret.Data).To(HaveKey("crd-verticalpodautoscalers.yaml"))
```

Comment (voelzmo) [reply pkg/component/vpa/vpa_test.go:1574]: Fair point, deleting the checks entirely went too far, I guess.

**Review by rfranzke — COMMENTED** [review]
Don't we also need to bump the images? https://github.com/gardener/gardener/blob/0db1e8ec215eb6d78b1e66b3e6b7ffaf7649e0ed/charts/images.yaml#L561-L600

**Comment by voelzmo** [comment]
> Don't we also need to bump the images?

For sure we do! Thanks for keeping an eye on this, seems like I was half asleep yesterday already. 🙈

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**